packages/cli/lib/util.js

```diff
@@ -54,3 +55,26 @@ exports.normalizeTemplatesResponse = function (repos = []) {
 exports.toBool = function (val) {
 	return val === void 0 || (val === 'false' ? false : val);
 };
+
+/**
+ * Taken from: https://github.com/preactjs/wmr/blob/3401a9bfa6491d25108ad68688c067a7e17d0de5/packages/wmr/src/lib/net-utils.js#L4-Ll4
+ * Check if a port is free
+ * @param {number} port
+ * @returns {Promise<boolean>}
+ */
+exports.isPortFree = async function (port) {
+	try {
+		await new Promise((resolve, reject) => {
+			const server = net.createServer();
+			server.unref();
+			server.on('error', reject);
+			server.listen({ port }, () => {
+				server.close(resolve);
+			});
+		});
+		return true;
+	} catch (err) {
+		if (err.code !== 'EADDRINUSE') throw err;
```

Comment:
this works cross platform?

Reply:
Should be. [Node's `server.listen()` docs](https://nodejs.org/api/net.html#serverlisten) list it as a common thing to check for, which would be odd if it wasn't cross platform.

I was under the impression that Node's error codes were all cross platform anyways. The docs don't seem to list anything on [this page](https://nodejs.org/api/errors.html#common-system-errors), (common system error or no), as being platform-specific. Could be wrong though?

Reply:
Found a random [StackOverflow question of someone getting it on Windows](https://stackoverflow.com/questions/41716833/getting-an-error-listen-eaddrinuse-3000-on-windows-machine/41717870), which I imagine would be the problem child if there was one.

Should be fine to check for that.